sapi/fpm/tests/tester.inc

```diff
@@ -268,6 +268,16 @@ class Tester
         }
     }
 
+    /**
+     * Skip if not running as root.
+     */
+    static public function skipIfNotRoot()
+    {
+        if (getmyuid() != 0) {
```

Comment:
This doesn't seem correct as it's just for the actual script owner but we should care about the EUID so it should be more something like this:
```php
        if (
            (extension_loaded('posix_geteuid') && posix_geteuid() !== 0) ||
            (exec('id -u', $output, $rv) !== "0")
        ) {
            die('skip not running as root');
        }
```

Reply:
Oh, I didn't know that's what `getmyuid` did. However the test still works as intended because when the test is run as root, the PHP script generated from the phpt file will be owned by root, and so `getmyuid` will return 0.

I can add a commit to replace the test as you suggested, but in `skipif.inc` there's a similar check that also uses `getmyuid`:

```
// Running as root is not allowed without TEST_FPM_RUN_AS_ROOT env
if (!getmyuid() && !getenv('TEST_FPM_RUN_AS_ROOT')) {
    die('skip Refusing to run as root');
}
```

Reply:
Ah yeah it should be enough then. I sometimes test things just by running php on the phpt file (without the framework) so that's why it didn't work for me but that's not requirement for the test to work so feel free to leave it as it is.